rtlib/src/ImageStitcher.cpp

```diff
@@ -0,0 +1,732 @@
+        filterMatchedFeatures(feature_img1, features1, features2);*/
+        dst_size = 0;
+        all_features_.push_back(img2_features);
+    }
+    else{
+        dst_size = all_features_[img2_idx].keypoints.size();
+        // Appends the new landmarks
+        for(int i = 0; i < ldm_pair.dst_ldms.size(); i++){
+            all_features_[img2_idx].keypoints.emplace_back(cv::KeyPoint(ldm_pair.dst_ldms[i].first, ldm_pair.dst_ldms[i].second, 31));
+        }
+        //May need to compute the descriptors
+    }
+
+    // Populate the matches and put them into pairwise matches
+    int src_dst_idx = search_matches_(ldm_pair.src_idx, ldm_pair.dst_idx);
+    cv::detail::MatchesInfo matches1 = all_pairwise_matches_[src_dst_idx];
+    for(int i = 0; i < ldm_pair.src_ldms.size(); i++){
+        matches1.matches.emplace_back(cv::DMatch(src_size + i, dst_size + i, 0));
+    }
+    compute_homography_(matches1);
+    all_pairwise_matches_[src_dst_idx] = matches1;
+
+    int dst_src_idx = search_matches_(ldm_pair.dst_idx, ldm_pair.src_idx);
+    cv::detail::MatchesInfo matches2 = all_pairwise_matches_[dst_src_idx];
+    for(int i = 0; i < matches1.matches.size(); i++){
+        matches2.matches.emplace_back(cv::DMatch(matches1.matches[i].trainIdx, matches1.matches[i].queryIdx, 0));
+    }
+    matches2.inliers_mask = matches1.inliers_mask;
+    matches2.confidence = matches1.confidence;
+    matches2.num_inliers = matches1.num_inliers;
+    matches2.H = matches1.H.inv();
+    all_pairwise_matches_[dst_src_idx] = matches2;
+}
+
+// This will probably need to be changed to account for having more than 2 images
+void ImageStitcher::setLandmarks(std::vector<std::string> ldmFiles){
+
+    if(ldmFiles.size() >= imgs_.size()){
+        throw std::runtime_error(
+                "Too many landmark files.");
+    }
+    for(int i = 0; i < ldmFiles.size(); i++){
+        LandmarkPair ldm;
+        ldm.src_idx = i;
+        ldm.dst_idx = i+1;
+        std::ifstream landmarkFile;
+        // Open the landmarks file
+        landmarkFile.open(ldmFiles[i]);
+        // Check that the landmarks file opened
+        if (!landmarkFile.is_open()) {
+            throw std::runtime_error(
+                    "Could not open landmarks file " + ldmFiles[i] + ".");
+        }
+
+        std::pair<float, float> point1;
+        std::pair<float, float> point2;
+        // Read in the landmarks and store them
+        landmarkFile >> point1.first;
+        while (!landmarkFile.eof()) {
+            landmarkFile >> point1.second;
+            ldm.src_ldms.push_back(point1);
+            landmarkFile >> point2.first;
+            landmarkFile >> point2.second;
+            ldm.dst_ldms.push_back(point2);
+            landmarkFile >> point1.first;
+        }
+        // Close the landmarks file
+        landmarkFile.close();
+        landmarks_.emplace_back(ldm);
+    }
+}
+
+std::vector<cv::detail::ImageFeatures> ImageStitcher::find_features_(double& work_scale_) {
```

Comment:
```suggestion
std::vector<cv::detail::ImageFeatures> ImageStitcher::find_features_(double workScale) {
```
Pass small constants by value.
Style: lowerCamelCase fn args.